client/src/client.ts

```diff
@@ -50,6 +51,45 @@ export class AngularLanguageClient implements vscode.Disposable {
       // Don't let our output console pop open
       revealOutputChannelOn: lsp.RevealOutputChannelOn.Never,
       outputChannel: this.outputChannel,
+      middleware: {
+        provideDefinition: async (
+            document: vscode.TextDocument, position: vscode.Position,
+            token: vscode.CancellationToken, next: lsp.ProvideDefinitionSignature) => {
+          if (document.languageId === 'typescript' &&
```

Comment:
What do you think if we simplify this to:
```ts
if (isInsideInlineTemplateRegion(document)) {
  next(document, position, token);
}
```
and do all the checks in the function?